internal/file-detector/detector.go

```diff
@@ -84,7 +84,7 @@ type fsNotifyOptions struct {
 
 // newFSNotify creates a new fsnotify detector which uses filesystem events to
 // detect that a file has changed.
-func newFSNotify(opts fsNotifyOptions) (*fsNotify, error) {
+func NewFSNotify(opts FsNotifyOptions) (*fsNotify, error) {
```

Comment:
nit: it's bad form for an exported function/method to return a non-exported type; you should export `fsNotify`